### PR TITLE
added a generator to data.random_split to make data splits consistent

### PR DESCRIPTION
**Pull request by eleanorplatt**
I added a generator (manual seed) to data.random_split to make data splits consistent across seeds. Previously the training and validation splits were different for each seed causing data leakage and irreproducibility. Documentation: https://pytorch.org/docs/stable/data.html#torch.utils.data.random_split

### CODE REVIEW

**Review by jack-willturner — COMMENTED** [review]
Could you pass this in as an argument to `load_dataset` with a default of 1 so that it's modifiable?

**Review by jack-willturner — APPROVED** [review]
